tests/testthat/test-assert-that.R

```diff
@@ -6,3 +6,11 @@ test_that("assert_that handles long false assertions gracefully", {
         "^isTRUE\\(.* [.]{3} is not TRUE$"
     )
 })
+
+test_that("assert_that handles has_name failures with multiple missing names", {
```

Comment:
added this test to catch the condition I mentioned above, where failures with a `call$which` that has length > 1 could produce a bad error message